content/en/user-guide/aws/kms/index.md

```diff
@@ -120,10 +120,127 @@ As with the previous `Encrypt` call, we have to get rid of the Base64-encoding t
 ```sh
 some important stuff
 ```
+Let's now create a hash-based message authentication code(HMAC) key and use it to generate and verify mac for a specified message. In this example we will use key specification as `HMAC_256` and mac algorithm as `HMAC_SHA_256`.
+
+An HMAC key can be created in KMS with the following command:
+
+{{< command >}} $ awslocal kms create-key --key-spec HMAC_256 --key-usage GENERATE_VERIFY_MAC {{</ command >}}
+
+Note that the `GENERATE_VERIFY_MAC` value for the `--key-usage` parameter is required even though it's the only valid value for HMAC KMS keys. The allowed values of `--key-spec` parameter can be checked in [official AWS documentation](https://docs.aws.amazon.com/kms/latest/developerguide/hmac.html#hmac-key-specs)
+
+The create HMAC command gives the following output:
+```sh
+{
+    "KeyMetadata": {
+        "AWSAccountId": "000000000000",
+        "KeyId": "922c14f0-f6e9-4d22-a56a-9619b78621f0",
+        "Arn": "arn:aws:kms:us-east-1:000000000000:key/922c14f0-f6e9-4d22-a56a-9619b78621f0",
+        "CreationDate": 1681215688.040106,
+        "Enabled": true,
+        "Description": "",
+        "KeyUsage": "GENERATE_VERIFY_MAC",
+        "KeyState": "Enabled",
+        "Origin": "AWS_KMS",
+        "KeyManager": "CUSTOMER",
+        "CustomerMasterKeySpec": "HMAC_256",
+        "KeySpec": "HMAC_256",
+        "MultiRegion": false,
+        "MacAlgorithms": [
+            "HMAC_SHA_256"
+        ]
+    }
+}
+```
+
+We can now use this key and a valid [MAC algorithm](https://docs.aws.amazon.com/kms/latest/developerguide/hmac.html#hmac-key-specs) to generate an HMAC for a message say "some important stuff". 
+
+{{< command >}} $ awslocal kms generate-mac --message "some important stuff" --key-id 922c14f0-f6e9-4d22-a56a-9619b78621f0 --mac-algorithm HMAC_SHA_256 --query Mac --output text | base64 --decode > my_encrypted_data {{</ command >}}
+
+The `GenerateMac` operation has the following output:
+```sh
+{
+    "Mac": "Kl1cQNOgdYmwm3zXCOBfBa/mgq7bywA2HhQj4lHazUg=",
+    "MacAlgorithm": "HMAC_SHA_256",
+    "KeyId": "arn:aws:kms:us-east-1:000000000000:key/922c14f0-f6e9-4d22-a56a-9619b78621f0"
+}
+```
+
+We now use `VerifyMac` which is done by computing an HMAC using the message, key, MAC algorithm and this is compared to the HMAC that you specify. . If the HMACs are identical, the verification succeeds; otherwise, it fails.
```

Comment:
```suggestion
We now use `VerifyMac` which works by computing an HMAC using the message, key, MAC algorithm and this is compared to the HMAC that you specify.
If the HMACs are identical, the verification succeeds; otherwise, it fails.
```